api/src/main/java/org/openmrs/module/reporting/report/renderer/ReportTemplateRenderer.java

```diff
@@ -228,7 +228,7 @@ public Object getReplacementValue(Object initialValue) {
 				replacementValue = new Integer(((Cohort) initialValue).size());
 			} 
 			else if (initialValue instanceof IndicatorResult) {
-				replacementValue = new Double(((IndicatorResult) initialValue).getValue().doubleValue());
+				replacementValue =((IndicatorResult) initialValue).getValue();
 			}

```

Comment:
The concern with this change is that it would be a breaking change for anyone currently relying upon the existing behavior.  What is your use case?

Reply:
@mseaton we are integrating with DHIS2, and some of the fields have an integer return value, which breaks when a double is used for example an indicator with value 2, returns 2.0 (any advice on how to do it differently?

Reply:
@ssmusoke makes sense.  Because most indicators (eg. CohortIndicators that have no denominator) are actually Integers, maybe we can just check datatype and handle appropriately first.  For example:

```
IndicatorResult is = (IndicatorResult)initialValue;
if (is.getValue() instanceof Integer) {
  replacementValue = (Integer)is.getValue();
}
else if (is.getValue() instanceof Double) {
  replacementValue = (Double)is.getValue();
}
else {
  new Double(((IndicatorResult) initialValue).getValue().doubleValue());
}
```

If you find you need to, you can also add more logic into the final else statement to check if the doubleValue is equal to a whole number and convert it to an Integer if so.

Reply:
@mseaton I have tried this suggestion, however, the line `IndicatorResult is = (IndicatorResult)initialValue;` throughs a class cast exception `java.lang.String cannot be cast to org.openmrs.module.reporting.indicator.IndicatorResult`

Reply:
Am now looking for a solution that works but using the implementation you suggested

Reply:
@mseaton and @ssmusoke I have implemented the solution as @mseaton suggested but with minor modifications. Here is what I did.

`if (initialValue != null) {
			 if (initialValue instanceof IndicatorResult) {
				IndicatorResult ir = (IndicatorResult) initialValue;
				if(ir.getValue() instanceof Integer)
				{
					replacementValue = (Integer) ir.getValue();
				}
				else if(ir.getValue() instanceof Double) {
					replacementValue = (Double) ir.getValue();
				}
			}
			 else {
			 	replacementValue = initialValue;
			 }
		}
		return replacementValue;`
Waiting for your comments